rules/go_rules.build_defs

```diff
@@ -100,12 +100,23 @@ def go_library(name:str, srcs:list, resources:list=None, asm_srcs:list=None, hdr
       package (str): The package as it would appear at the top of the go source files. Defaults
                      to name.
     """
+    cover = cover and CONFIG.BUILD_CONFIG == "cover"
     package = package or name
     out = package + '.a'
     link_labels = []
     src_labels = []
     private = out.startswith('_')
 
+    # We could just depend on go_src for all our deps but that would mean bringing in srcs for targets outside this
+    # package like third party sources which is especially slow on systems with slow syscall performance (macOS)
+    if _all_srcs:
+        deps += [filegroup(
+            name = name,
+            tag = "package_srcs",
+            exported_deps = [dep for dep in deps if dep.startswith(":")],
```

Comment:
I wonder if we should implement some built in functions to help with this sort of thing. I know build rules ideally wouldn't have to worry about this stuff but it's evident to me that it's useful in some cases.